js/src/components/paid-ads/budget-section/index.js

```diff
@@ -13,30 +14,70 @@ import BudgetRecommendation from './budget-recommendation';
 import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
 import AppInputPriceControl from '.~/components/app-input-price-control';
 
+const nonInteractableProps = {
+	noPointerEvents: true,
+	readOnly: true,
+	tabIndex: -1,
+};
+
+/**
+ * Renders <Section> and <Section.Card> UI with campaign budget inputs.
+ *
+ * @param {Object} props React props.
+ * @param {Object} props.formProps Form props forwarded from `Form` component.
+ * @param {boolean} [props.disabled=false] Whether display the Card in disabled style.
+ */
 const BudgetSection = ( props ) => {
 	const {
-		formProps: { getInputProps, values },
+		formProps: { getInputProps, setValue, values },
+		disabled = false,
 	} = props;
 	const {
 		country: [ selectedCountryCode ],
 		amount,
 	} = values;
 	const { googleAdsAccount } = useGoogleAdsAccount();
-	const monthlyMaxEstimated = getMonthlyMaxEstimated( values.amount );
+	const monthlyMaxEstimated = getMonthlyMaxEstimated( amount );
 	// Display the currency code that will be used by Google Ads, but still use the store's currency formatting settings.
 	const currency = googleAdsAccount?.currency;
 
+	/**
+	 * In addition to the initial value setting during initialization, when `disabled` changes
+	 * - from false to true, then clear filled amount to `undefined` for showing a blank <input>.
+	 * - from true to false, then reset amount to the initial value passed from the consumer side.
+	 */
+	const initialAmountRef = useRef( amount );
+	useEffect( () => {
+		const nextAmount = disabled ? undefined : initialAmountRef.current;
+		setValue( 'amount', nextAmount );
+	}, [ disabled, setValue ] );
```

Comment:
❓ Interesting. There are two things I like to touch on: 

The first one is about `initialAmountRef` and `nextAmount`. They are used to "reset amount to the initial value passed from the consumer side". When I read https://github.com/woocommerce/google-listings-and-ads/pull/1268#discussion_r813246789 (with @eason9487 and @j111q), my interpretation of "reset" is to make the value undefined (make it blank), and one reason for it is that when it is disabled, the input box is blank. If we were to reset it to initial value (i.e. enabled with initial value "5", then disabled with blank value, then enabled with initial value "5" again), the disappearing and re-appearing of value feel weird for me. 😅 

The second one is about `useEffect`. Instead of using `useEffect`, my thought is to tap into the SelectControl's `onChange` event handler inside `AudienceSection` component, i.e. in `onChange`, if the new value is empty array (no country), then we call `setValue( 'amount', undefined )`. What do you think? 🤔

Reply:
> If we were to reset it to initial value (i.e. enabled with initial value "5", then disabled with blank value, then enabled with initial value "5" again), the disappearing and re-appearing of value feel weird for me.

It still depends on scenarios. Usually, forms are set to initial/default values when "reset" and set to blank/default values when "clear". And from an input component design perspective, it should use the given initial value. Otherwise, as a developer, it looks like a bug if the initial value on \<Form\> is `5` and the input shows blank rather than the given initial one when changing from disabled to enabled.

> The second one is about `useEffect`. Instead of using `useEffect`, my thought is to tap into the SelectControl's `onChange` event handler inside `AudienceSection` component, i.e. in `onChange`, if the new value is empty array (no country), then we call `setValue( 'amount', undefined )`. What do you think? 🤔

Yes, it's better to reset amount relying on the changing of country values, but resetting amount in country selector will increase coupling between these components. I don't think it's a good idea.

Reply:
> resetting amount in country selector will increase coupling between these components

Yeah, I think I get what you mean, I wish there is a way to put that logic in the form component level and not in the `AudienceSection`, but I don't think there is one.

I guess we can go with what we have here.